server/src/main/java/org/opensearch/indices/replication/common/ReplicationLuceneIndex.java

```diff
@@ -121,6 +122,13 @@ public synchronized int totalFileCount() {
         return filesDetails.size();
     }
 
+    /**
+     * Returns true if set of files represent an incoming greater commit point
```

Comment:
nit - the word 'greater' is misleading here, there is no validation within this class that indicates that, only the presence of a segments_N.